lib/rules/no-implicit-this.js

```diff
@@ -52,7 +53,7 @@ const ARGLESS_DEFAULT_BLUEPRINT = [
 
 export default class NoImplicitThis extends Rule {
   parseConfig(config) {
-    if (config === false || config === undefined) {
+    if (config === false || config === undefined || isSupportedScriptFileExtension(this.filePath)) {
```

Comment:
Hmm, I don't think this is quite right. We don't care if it is in a `.js`/`.gjs` file we care if the usage was via a known strict mode system like `<template>`.  There is nothing that says you can't use `import { hbs }from 'ember-cli-htmlbars'` inside a `.gjs` file.

This is getting back to the same thing I was referring to above (and requires https://github.com/ember-template-imports/ember-template-imports/issues/42 to be completed in order to resolve).

I think we'll need to complete https://github.com/ember-template-imports/ember-template-imports/issues/42 and then add an additional property on the `Rule` base class. Something like `this.isStrictMode` or somesuch...

Reply:
Let's consolidate the conversation about https://github.com/ember-template-imports/ember-template-imports/issues/42 in [this thread](https://github.com/ember-template-lint/ember-template-lint/pull/2483#discussion_r938027129).